modules/sor/sorV2/lib/pathGraph/pathGraph.ts

```diff
@@ -457,38 +506,75 @@ export class PathGraph {
         return filtered;
     }
 
-    private getLimitAmountSwapForPath(path: PathGraphEdgeData[], swapKind: SwapKind): bigint {
-        let limit = path[path.length - 1].pool.getLimitAmountSwap(
-            path[path.length - 1].tokenIn,
-            path[path.length - 1].tokenOut,
-            swapKind,
-        );
-
-        for (let i = path.length - 2; i >= 0; i--) {
-            const poolLimitExactIn = path[i].pool.getLimitAmountSwap(
-                path[i].tokenIn,
-                path[i].tokenOut,
+    private getLimitAmountSwapForPath(path: PathGraphEdgeData[]): bigint {
+        let limit;
+        if (this.getOperationFromPath(path[path.length - 1]) === PathOperation.AddLiquidity) {
+            limit = path[path.length - 1].pool.getLimitAmountAddLiquidity(path[path.length - 1].tokenIn);
+        } else if (this.getOperationFromPath(path[path.length - 1]) === PathOperation.RemoveLiquidity) {
+            limit = path[path.length - 1].pool.getLimitAmountRemoveLiquidity();
+        } else {
+            limit = path[path.length - 1].pool.getLimitAmountSwap(
+                path[path.length - 1].tokenIn,
+                path[path.length - 1].tokenOut,
                 SwapKind.GivenIn,
             );
-            const poolLimitExactOut = path[i].pool.getLimitAmountSwap(
-                path[i].tokenIn,
-                path[i].tokenOut,
-                SwapKind.GivenOut,
-            );
+        }
 
-            if (poolLimitExactOut <= limit) {
-                limit = poolLimitExactIn;
+        for (let i = path.length - 2; i >= 0; i--) {
+            let limitGivenIn;
+            let limitGivenOut;
+            if (this.getOperationFromPath(path[i]) === PathOperation.Swap) {
+                limitGivenIn = path[i].pool.getLimitAmountSwap(path[i].tokenIn, path[i].tokenOut, SwapKind.GivenIn);
+                limitGivenOut = path[i].pool.getLimitAmountSwap(path[i].tokenIn, path[i].tokenOut, SwapKind.GivenOut);
+                if (limitGivenOut <= limit) {
+                    limit = limitGivenIn;
+                } else {
+                    const pulledLimit: bigint = path[i].pool.swapGivenOut(
+                        path[i].tokenIn,
+                        path[i].tokenOut,
+                        TokenAmount.fromRawAmount(path[i].tokenOut, limit),
+                    ).amount;
+                    limit = pulledLimit > limitGivenIn ? limitGivenOut : pulledLimit;
```

Comment:
That's what was on the previous code. It makes more sense to me. Did you change it intentionally?
```suggestion
                    limit = pulledLimit > limitGivenIn ? limitGivenIn : pulledLimit;
```